interfaces/repo.go

```diff
@@ -470,25 +488,15 @@ func (r *Repository) securitySnippetsForSnap(snapName string, securitySystem Sec
 		if err != nil {
 			return nil, err
 		}
-		if snippet != nil {
-			for appName := range slot.Apps {
-				securityTag := snap.AppSecurityTag(snapName, appName)
-				snippets[securityTag] = append(snippets[securityTag], snippet)
-			}
-		}
+		addSnippet(snapName, slot.Name, slot.Apps, nil, snippets, snippet)
+
 		// Add connection-specific snippet specific to each plug
 		for plug := range r.slotPlugs[slot] {
 			snippet, err := iface.ConnectedSlotSnippet(plug, slot, securitySystem)
 			if err != nil {
 				return nil, err
 			}
-			if snippet == nil {
-				continue
-			}
-			for appName := range slot.Apps {
-				securityTag := snap.AppSecurityTag(snapName, appName)
-				snippets[securityTag] = append(snippets[securityTag], snippet)
-			}
+			addSnippet(snapName, slot.Name, slot.Apps, nil, snippets, snippet)
```

Comment:
Right, it comes from the snap slot as far as this side of the connection is concerned.